tests/js/server/aql/aql-memory-limit.js

```diff
@@ -95,16 +87,226 @@ function ahuacatlMemoryLimitTestSuite () {
-    }
+    },
 
   };
 }
 
-////////////////////////////////////////////////////////////////////////////////
-/// @brief executes the test suite
-////////////////////////////////////////////////////////////////////////////////
+function ahuacatlMemoryLimitReadOnlyQueriesTestSuite () {
+  const cn = "UnitTestsCollection";
 
-jsunity.run(ahuacatlMemoryLimitTestSuite);
+  let c;
 
-return jsunity.done();
+  return {
+    setUpAll : function () {
+      // only one shard because that is more predictable for memory usage
+      c = db._create(cn, { numberOfShards: 1 });
+
+      let docs = [];
+      for (let i = 0; i < 100 * 1000; ++i) {
+        docs.push({ value1: i, value2: i % 10, _key: "test" + i });
+        if (docs.length === 5000) {
+          c.insert(docs);
+          docs = [];
+        }
+      }
+    },
+    
+    tearDownAll : function () {
+      db._drop(cn);
+    },
+
+    testFullScan : function () {
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " RETURN doc", null, { memoryLimit: 10 * 1000 * 1000 }).json;
+      assertEqual(100000, actual.length);
+        
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " RETURN doc", null, { memoryLimit: 5 * 1000 * 1000 });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+    
+    testIndexScan : function () {
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " SORT doc._key RETURN doc", null, { memoryLimit: 10 * 1000 * 1000 }).json;
+      assertEqual(100000, actual.length);
+        
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " SORT doc._key RETURN doc", null, { memoryLimit: 5 * 1000 * 1000 });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+
+    testSort : function () {
+      // turn off constrained heap sort
+      const optimizer = { rules: ["-sort-limit"] };
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " SORT doc.value1 LIMIT 10 RETURN doc", null, { memoryLimit: 15 * 1000 * 1000, optimizer }).json;
+      assertEqual(10, actual.length);
+        
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " SORT doc.value1 LIMIT 10 RETURN doc", null, { memoryLimit: 10 * 1000 * 1000, optimizer });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+    
+    testCollect1 : function () {
+      let actual = AQL_EXECUTE("FOR doc IN " + cn + " COLLECT v = doc.value1 OPTIONS { method: 'hash' } RETURN v", null, { memoryLimit: 10 * 1000 * 1000 }).json;
+      assertEqual(100000, actual.length);
+      
+      try {
+        AQL_EXECUTE("FOR doc IN " + cn + " COLLECT v = doc.value1 OPTIONS { method: 'hash' } RETURN v", null, { memoryLimit: 5 * 1000 * 1000 });
+        fail();
+      } catch (err) {
+        assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
+      }
+    },
+    
+    testCollect2 : function () {
```

Comment:
You could add some note here, as it is not directly clear for me where the difference between `testCollect1` and `testCollect2` is. First one uses `doc.value1`, the other one `doc.value2`. Both seem to be numeric values.